src/app/loadout-builder/hooks/useProcess.ts

```diff
@@ -219,8 +234,9 @@ function groupItems(
     }
 
     if (
-      someModHasEnergyRequirement(lockedArmor2ModMap.seasonal) ||
-      someModHasEnergyRequirement(lockedArmor2ModMap[ModPickerCategories.general])
+      !ignoreAffinity &&
```

Comment:
I don't think we want this here, it will have large ramifications on how items are grouped and what kind of sets we see as a result. The only time we would want this is if we blanket ignore elemental affinity, but from my understanding we ignore it after a threshold.

Reply:
Should this be cut out completely or changed to match `filterItems()` with `&& isEnergyLower(item, maximumEnergy`

Reply:
Cut out completely I believe, at this stage we don't care what items are doing we only care if the mods themselves have an energy requirement. This flows into the process worker. I think you would want do something similar to the filter in process.ts but I would need to go digging to say for sure.

Reply:
Maybe in these two places

https://github.com/DestinyItemManager/DIM/blob/c4d36f3f9accd17a74b3c453a990d967a02016af/src/app/loadout-builder/processWorker/processUtils.ts#L185
https://github.com/DestinyItemManager/DIM/blob/c4d36f3f9accd17a74b3c453a990d967a02016af/src/app/loadout-builder/processWorker/processUtils.ts#L206

Reply:
Just as an FYI there are lots of things around this that have flow on effects starting in the useProcess hook that travel down pretty deep. Performance was a real issue with all this stuff so it was prioritised above all else.

Reply:
Seems like this was unnecessary and the change in `preProcessFilter/filterItems` is doing the job.